recipes/remove-whitespace-from-string.md

```diff
@@ -0,0 +1,63 @@
+# Remove whitespace from string
+
+## Problem
+
+Remove extraneous whitespace (i.e. space, tab, newline) characters
+from a string.
+
+## Solution
+
+### Using SRFI 13
+
+These can remove arbitrary characters, but the default is to remove
+whitespace:
+
+* `(string-trim s)`
+* `(string-trim-right s)`
+* `(string-trim-both s)`
+
+### Using regular expressions
+
+```
+(define string-trim-both
+  (let ((r (regexp "^[ \t\r]*(.*?)[ \t\r]*$")))
+    (lambda (s)
+      (cadr (regexp-match r s)))))
```

Comment:
From where `regexp` and `regexp-match` came from? Is it something specific to one implementation or part of SRFI? I don't see this in SRFI-13. It looks implementation-specific for the old cookbook.

Reply:
Ah, they're from Racket (possibly in other Schemes too). I'll push another commit to mention it.

Reply:
Pushed.

Reply:
I don't think it's a good idea to have something from Racket or any single implementation. It would be ideal if the solution works for most implementation. Note that there is SRFI-115 that implements regular expressions in form of S-Expressions, I still prefer to use strings like regexes but the standard Scheme regex is the one in SRFI (but I'm not sure how many implementations have support for it). I personally like regex literals that are implemented in Gauche (and I have the same syntax in LIPS).